### DISCUSSION TITLE
Ports are not updated on restart?

### DISCUSSION THREAD

**Discussion by zb226** [fail2ban/fail2ban#3511] [Q&A] [answered]
Hi and thanks for this great tool! I have set up a rule for the mailserver maddy and I added ports to the configuration in `jail.local`, from

```
[maddy-auth]
port     = 993,465,25
```
to

```
[maddy-auth]
port     = 993,587,465,143,25
```

In order to have the ports updated I thought I'd need to run `fail2ban-client restart`, but after doing so, there's still the old ports active in the firewall:

```
Chain INPUT (policy ACCEPT)
target     prot opt source               destination
f2b-sshd   tcp  --  0.0.0.0/0            0.0.0.0/0            multiport dports 22
f2b-maddy-auth  tcp  --  0.0.0.0/0            0.0.0.0/0            multiport dports 993,465,25
```

What am I doing wrong?

**Comment by sebres** [comment, marked as answer]
Basically impossible, at least with almost neuest versions of fail2ban. 
But so if one does really `fail2ban-client restart` (not `fail2ban-client reload`), because `restart` in opposite to `reload` completely restarts the server (or if used for single jail it recreates the jail and rebans everything), whereas `reload` would only update the runtime configuration, but wouldn't restart the actions and doesn't reban the IPs.
To ensure what configuration would reach the server, check the port in output of this:
```
fail2ban-client -d | grep maddy-auth
```
So my impression is either you did use `reload` (use `fail2ban-client restart maddy-auth` instead) or something is not quite correct in the configuration.

**Reply by zb226** [reply]
Thanks for answering! I'm sorry, I mixed something up in my question. The above `iptables` output is actually from when `fail2ban` was stopped. I got confused because somehow I managed that the old `iptables` rule was retained. When I started `fail2ban` again, I got:

```
Chain INPUT (policy ACCEPT)
target     prot opt source               destination
f2b-maddy-auth  tcp  --  0.0.0.0/0            0.0.0.0/0            multiport dports 993,587,465,143,25
f2b-sshd   tcp  --  0.0.0.0/0            0.0.0.0/0            multiport dports 22
f2b-maddy-auth  tcp  --  0.0.0.0/0            0.0.0.0/0            multiport dports 993,465,25
```

I've stopped `fail2ban`, deleted that rule manually and after starting it again all seems fine:

```
Chain INPUT (policy ACCEPT)
target     prot opt source               destination
f2b-maddy-auth  tcp  --  0.0.0.0/0            0.0.0.0/0            multiport dports 993,587,465,143,25
f2b-sshd   tcp  --  0.0.0.0/0            0.0.0.0/0            multiport dports 22
```
